storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -283,9 +346,22 @@ func (s *fileSystemSource) handleErrorsAndUnfulfilled(
 	}
 }
 
+func (s *fileSystemSource) tagsFromTagsIter(
+	iter ident.TagIterator,
+) (ident.Tags, error) {
+	tags := make(ident.Tags, 0, iter.Remaining())
+	for iter.Next() {
+		curr := iter.Current()
+		tags = append(tags, s.idPool.CloneTag(curr))
+	}
+	return tags, iter.Err()
+}
+
 func (s *fileSystemSource) loadShardReadersDataIntoShardResult(
+	ns namespace.Metadata,
```

Comment:
can I convince you to break this function into smaller chunks? it was huge to start, now with both code-paths (index/data) in the same function, it's really un-wieldly

Reply:
Sure I can break some of it.